File: server/src/main/java/com/fluffy/exam/domain/dto/MyExamSummaryDto.java
package com.fluffy.exam.domain.dto;

import com.fluffy.exam.domain.ExamStatus;
import com.querydsl.core.annotations.QueryProjection;
import java.time.LocalDateTime;

public record MyExamSummaryDto(
        Long id,
        String title,
        String description,
        ExamStatus status,
        AuthorDto author,
        Long questionCount,
        LocalDateTime createdAt,
        LocalDateTime updatedAt
) {

    @QueryProjection
    public MyExamSummaryDto {
    }
}

File: server/src/main/kotlin/com/fluffy/exam/domain/dto/MyExamSummaryDto.kt
package com.fluffy.exam.domain.dto

import com.fluffy.exam.domain.ExamStatus
import java.time.LocalDateTime

data class MyExamSummaryDto(
    val id: Long,
    val title: String,
    val description: String,
    val status: ExamStatus,
    val author: AuthorDto,
    val questionCount: Long,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime,
)
